src/Compilers/CSharp/Portable/Binder/PatternSwitchBinder.cs

```diff
@@ -25,38 +26,134 @@ internal PatternSwitchBinder(Binder next, SwitchStatementSyntax switchSyntax) :
+                switchSections: switchSections,
+                // If there is no explicit default label, the default action is to break out of the switch
+                defaultLabel: defaultLabel?.Label ?? BreakLabel,
+                diagnostics);
+
+            // Report subsumption errors, but ignore the input's constant value for that.
+            CheckSwitchErrors(node, boundSwitchGoverningExpression, ref switchSections, decisionDag, diagnostics);
+
+            if (boundSwitchGoverningExpression.ConstantValue != null)
+            {
+                // When the input is constant, we use that to reshape the decision dag that is returned
+                // so that flow analysis will see that some of the cases may be unreachable.
+                decisionDag = decisionDag.SimplifyDecisionDagForConstantInput(boundSwitchGoverningExpression, Conversions, diagnostics);
+            }
+
             return new BoundPatternSwitchStatement(
-                node, boundSwitchGoverningExpression, someCaseMatches,
-                locals, functions, switchSections, defaultLabel, this.BreakLabel, isComplete);
+                syntax: node,
+                expression: boundSwitchGoverningExpression,
+                innerLocals: locals,
+                innerLocalFunctions: functions,
+                switchSections: switchSections,
+                defaultLabel: defaultLabel,
+                breakLabel: this.BreakLabel,
+                decisionDag: decisionDag);
+        }
+
+        private static void CheckSwitchErrors(
+            SwitchStatementSyntax node,
+            BoundExpression boundSwitchGoverningExpression,
+            ref ImmutableArray<BoundPatternSwitchSection> switchSections,
+            BoundDecisionDag decisionDag,
+            DiagnosticBag diagnostics)
+        {
+            HashSet<LabelSymbol> reachableLabels = decisionDag.ReachableLabels;
+            bool isSubsumed(BoundPatternSwitchLabel switchLabel)
+            {
+                return !reachableLabels.Contains(switchLabel.Label);
+            }
+            bool areAnySubsumed(ImmutableArray<BoundPatternSwitchSection> sections)
+            {
+                foreach (BoundPatternSwitchSection section in sections)
+                {
+                    foreach (BoundPatternSwitchLabel label in section.SwitchLabels)
+                    {
+                        if (!label.HasErrors && isSubsumed(label))
+                        {
+                            // we found a label that is subsumed
+                            return true;
+                        }
+                    }
+                }
+
+                return false;
+            }
+
+            if (!areAnySubsumed(switchSections))
+            {
+                return;
+            }
+
+            // We mark any subsumed sections as erroneous for the benefit of flow analysis
+            var sectionBuilder = ArrayBuilder<BoundPatternSwitchSection>.GetInstance();
```

Comment:
Hint size?

Reply:
Recorded in [https://github.com/dotnet/roslyn/issues/25663](https://github.com/dotnet/roslyn/issues/25663)

---
In reply to: [176223051](https://github.com/dotnet/roslyn/pull/24059#discussion_r176223051) [](ancestors = 176223051)